integreat_cms/cms/views/pages/page_form_view.py

```diff
@@ -495,6 +511,65 @@ def post(self, request, *args, **kwargs):
             },
         )
 
+    # pylint: disable=consider-using-assignment-expr
+    @staticmethod
+    def cleanup_autosaves(objects, language, keep=3):
+        """
+        Delete all but the last 3 AUTO_SAVEs
+        and renumber all affected versions
+        to be continuous.
+        """
+        all_versions = objects.filter(language=language)
+        auto_saves = all_versions.filter(status=status.AUTO_SAVE)
+        versions = list(reversed(list(map(lambda x: x.version, auto_saves))))
+        if not versions:
+            # nothing to clean up
+            return 0
+        earliest = versions[0]
+        affected_versions = all_versions.filter(version__gte=earliest)
+        try:
+            lastN = versions[-keep]
+            to_delete = auto_saves.filter(version__lt=lastN)
+            v_deleted = list(reversed(list(map(lambda x: x.version, to_delete))))
+            print(f"##### v_deleted = {v_deleted}")
+            to_delete.delete()
+            num_deleted = len(v_deleted)
+
+            to_rename = affected_versions.exclude(version__in=v_deleted)
+            print(
+                f"##### to_rename = {list(reversed(list(map(lambda x: x.version, to_rename))))}"
+            )
+            # Build a lookup table to fix version numbers,
+            # which are ascendeing inteǵers without skips
+            mapping = {}
+            for i in range(earliest, max(map(lambda x: x.version, to_rename)) + 1):
+                if i in v_deleted:
+                    # version was deleted
+                    mapping[i] = mapping[i - 1] if i != earliest else earliest - 1
+                else:
+                    mapping[i] = mapping[i - 1] + 1 if i != earliest else earliest
+
+            for v in to_rename:
+                v.version = mapping[v.version]
```

Comment:
Not sure, if I'm missing something, but wouldn't something like 
```
to_rename.order_by("version")
for i, version_instance in enumerate(to_rename): 
    version_instance.version = earliest + i
```
do the same here?

Reply:
Yes, good idea! I think this is much better to understand. We could actually write
```
for new_version, translation in enumerate(to_rename, start=earliest):
    translation.version = new_version
```
to improve it even more.